core/trino-main/src/main/java/io/trino/server/DynamicFilterService.java

```diff
@@ -333,6 +347,14 @@ public void stageCannotScheduleMoreTasks(StageId stageId, int numberOfTasks)
         executor.submit(() -> collectDynamicFilters(stageId, Optional.empty()));
     }
 
+    public static Set<DynamicFilterId> getInterStageConsumedDynamicFilters(PlanFragment plan)
```

Comment:
Call it `getPropagatedDynamicFilters` (plus javadoc) or something similar. Current name is too specific and `InterStageConsumedDynamic` is really a detail.

Also, I already used `interStageDynamicFilters` within `getLazyDynamicFilters`